packages/terra-data-grid/src/FlowsheetDataGrid.jsx

```diff
@@ -157,6 +157,35 @@ function FlowsheetDataGrid(props) {
     isResizable: false,
   })), [columns]);
 
+  const tableBodyColumns = useMemo(() => {
+    let i = 0;
+    const newDisplayedColumns = [];
+    flowsheetColumns.forEach((column, columnHeaderIndex) => {
+      if (column.columnSpan > 1) {
+        newDisplayedColumns[i] = {
+          ...column,
+          columnSpanIndex: 0,
+          columnHeaderIndex,
+          columnHighlightColor: undefined,
+        };
+        i += 1;
+        for (let counter = column.columnSpan; counter > 1; counter -= 1) {
+          newDisplayedColumns[i] = {
+            ...column,
+            columnSpanIndex: (column.columnSpan - counter + 1),
+            columnHeaderIndex,
+            columnHighlightColor: undefined,
+          };
+          i += 1;
+        }
+      } else {
+        newDisplayedColumns[i] = { ...column, columnHeaderIndex };
+        i += 1;
+      }
```

Comment:
You should be able to handle all of this in a single for loop.  If the column span is undefined, you could just initialize the value to 1 for your test condition.  Then you would not require an if/else statement or have to set a value before entering the loop.  You can also use the push function to add values to your array so that you don't have to manually track the index.

Reply:
Updated.